decentraland/unity-renderer
# fix: collider bounds in scene bounds check

### WHY
There's a scene at [90,-143](https://play.decentraland.org/?position=90%2C-143) that has an entity with huge mesh outside the scene boundaries and that mesh doesn't have any renderers, it only has colliders.

We weren't taking colliders into consideration for the entities merged bounds calculation, and when there were no renderers in the entity we weren't affecting its colliders (since the entity was treated as an entity without a mesh).

### WHAT
- Added collider bounds to entity merged bounds calculation and scene bounds checks
- Added Scene boundaries check optimization: escape unneeded checks after outer bounds checks on entity addition
- Changed `MeshesInfo.colliders` to be a `HashSet<Collider>` instead of a `List<Collider>`
- Improved debug method in parcel scene
- Added scene boundaries checker test for GLTFshapes with only colliders and no renderers
- Added `DCLTransform` values capping
- Updated `ParametrizedShape` visibility configuration to take into account entitie's inside-scene-boundaries state 
- Updated `TextShape` visibility configuration to take into account entitie's inside-scene-boundaries state
- Added red wireframe effect to colliders outside scene boundaries for preview mode + scene boundaries checker debug mode test

### TESTING
1. Check that the reported issue got solved:
In [production](https://play.decentraland.org/?position=90%2C-143) there are invisible colliders on the empty parcels at the left of the affected scene (90, -143).
In [this PR's build](https://play.decentraland.zone/?renderer-branch=fix/ColliderBoundsInSceneBoundsCheck&position=90%2C-143) those colliders are disabled as they are detected as part of the mesh of the entity that contains them and that is detected as outside the scene boundaries.

2. Check that the already-solved https://github.com/decentraland/unity-renderer/issues/2132 issue hasn't reappeared:
    1. Spawn [at -12, -10](https://play.decentraland.zone/?renderer-branch=fix/ColliderBoundsInSceneBoundsCheck&position=-12%2C-10) and wait until Genesis Plaza finishes loading (make sure the Scene Load Radius value in the settings is 4) 
    2. Open Settings, change the Scene Load Radius to 1 and wait 5~ seconds
    3. Change Scene Load Radius to 4 and close the settings panel, wait for 5~10 seconds after the Genesis Plaza finishes loading and notice that he "black and white" bug doesn't happen anymore
    4. Steps 2 and 3 can be reproduced many times and the black and white bug won't happen again 
[Doing those steps from master's build will trigger the black and white bug]

3. Check that the already-solved https://github.com/decentraland/unity-renderer/issues/844 issue hasn't reappeared:
    1. Spawn [at -12, -10](https://play.decentraland.zone/?renderer-branch=fix/ColliderBoundsInSceneBoundsCheck&position=-12%2C-10)
    2. Walk towards the tramline at -12, 0
    3. Walk through the tramline to the west, and notice that no mesh is instantiated int the middle of the street
